agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -279,42 +279,79 @@ private void log(Logger logger) {
             Moshi moshi = MoshiBuilderFactory.createBuilderWithAdaptor();
-            JsonAdapter<Configuration> jsonAdapter = moshi.adapter(Configuration.class).failOnUnknown();
+            JsonAdapter<Configuration> jsonAdapter = strict ? moshi.adapter(Configuration.class).failOnUnknown() :
+                    moshi.adapter(Configuration.class);
             Buffer buffer = new Buffer();
             buffer.readFrom(in);
             try {
                 Configuration configuration = jsonAdapter.fromJson(buffer);
-                if (configuration.instrumentationSettings != null) {
-                    throw new IllegalStateException("It looks like you are using an old applicationinsights.json file" +
-                            " which still has \"instrumentationSettings\", please see the docs for the new format:" +
-                            " https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config");
-                }
-                configuration.configPath = configPath;
-                configuration.lastModifiedTime = lastModifiedTime;
                 return configuration;
             } catch(JsonDataException ex) {
-                throw new FriendlyException("Application Insights Java agent's configuration file "+configPath.toAbsolutePath().toString()+" has the following issue:\n"+ex.getMessage(),
-                        "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358");
-
+                Logger logger = LoggerFactory.getLogger(ConfigurationBuilder.class);
+                logger.warn(ex.getMessage());
+                if(strict) {
+                    // Try extracting the configuration without failOnUnknown
+                    return getConfigurationFromConfigFile(configPath, false);
+                } else {
+                    throw new FriendlyException("Application Insights Java agent's configuration file "+configPath.toAbsolutePath().toString()+" has the following json issue:\n"+ex.getMessage(),
+                            "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358");
+                }
             } catch (JsonEncodingException ex) {
-                throw new FriendlyException("Application Insights Java agent's configuration file "+configPath.toAbsolutePath().toString()+" has the following syntax issue:\n"+ex.getMessage(),
+                throw new FriendlyException(getJsonEncodingExceptionMessage(configPath.toAbsolutePath().toString(), ex.getMessage()),
                         "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358");
-
             } catch(Exception e) {
                 throw new ConfigurationException("Error parsing configuration file: " + configPath.toAbsolutePath().toString(), e);
             }
         }
     }
+
+    static String getJsonEncodingExceptionMessage(String configPath, String message) {
```

Comment:
I added this to make the error messages more clear to the customers. Moshi's exception messages are not very relevant to our usecases.